source/PlayerInfo.cpp

```diff
@@ -3165,6 +3198,22 @@ void PlayerInfo::SelectShip(const shared_ptr<Ship> &ship, bool *first)
 
 
 
+// When we remove a ship, forget it's stored Uuid.
+void PlayerInfo::ForgetShip(const EsUuid &uuid)
+{
+	string name = "";
+	for(const auto &ship : giftedShips)
+		if(ship.second == uuid)
+		{
+			name = ship.first;
+			break;
+		}
+	if(!name.empty())
```

Comment:
I would expect `find_if` with a reference capture of the id instead of copying a string and then iterating again.